### PR TITLE
Remove nodejs dependencies, add slightly better validation

### PR DESCRIPTION
**Pull request by SimonSchick**
This is a byproduct of trying to fix AOT builds and might not be required anymore, don't want to drop dead this work tho.
- replacing the single querystring call with some basic forward validation
- replacing `events` with a barebone implementation
- updated dependencies
- strictened up TS (fixing potential bugs)
- module target is now es2015 for tree shaking
- added error metadata for better grouping

### CODE REVIEW

**Comment by SimonSchick** [comment]
Ignore breaking build, carina tests are flaky...

**Comment by ProbablePrime** [comment]
As talked about in slack not a fan of removing `events` here

File: tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,16 @@
 {
     "compilerOptions": {
-        "module": "commonjs",
+        "declaration": true,
+        "experimentalDecorators": true,
+        "emitDecoratorMetadata": true,
```
Comment (jamesbirtles) [inline tsconfig.json]: Neither of these are needed

File: webpack.config.js [hunk 2]
```diff
@@ -10,11 +10,11 @@ module.exports = {
         libraryTarget: 'umd',
     },
     resolve: {
-        extensions: ['', '.webpack.js', '.web.js', '.ts', '.js'],
+        extensions: ['.webpack.js', '.web.js', '.ts', '.js'],
     },
     module: {
         loaders: [
-            { test: /\.json$/, loaders: ['json'] },
+            { test: /\.json$/, loaders: ['json-loader'] },
```
Comment (jamesbirtles) [inline webpack.config.js]: you can actually just remove this line. webpack 2 deals with json by default

File: webpack.config.js [hunk 3]
```diff
@@ -10,11 +10,11 @@ module.exports = {
         libraryTarget: 'umd',
     },
     resolve: {
-        extensions: ['', '.webpack.js', '.web.js', '.ts', '.js'],
+        extensions: ['.webpack.js', '.web.js', '.ts', '.js'],
```
Comment (jamesbirtles) [inline webpack.config.js]: might as well just make this `['.js', '.ts']`

File: src/events.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
+
```
Comment (jamesbirtles) [inline src/events.ts]: 👎 No need to write our own here

**Review by jamesbirtles — CHANGES_REQUESTED** [review]

File: src/events.ts (hunk 4 above)
Comment (SimonSchick) [reply src/events.ts]: Can we avoid it being bundled like 10x in fe2 then?

Comment (jamesbirtles) [reply src/events.ts]: Maybe? I don't know what the issue is, but doing this just means we can't ever optimise there

**Review by jamesbirtles — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -26,15 +26,16 @@
   },
   "homepage": "https://github.com/WatchBeam/carina#readme",
   "devDependencies": {
-    "awesome-typescript-loader": "^2.2.4",
+    "awesome-typescript-loader": "^3.0.3",
     "chai": "^3.5.0",
     "chai-subset": "^1.3.0",
-    "json-loader": "^0.5.4",
+    "cross-env": "^3.1.4",
```
Comment (connor4312) [inline package.json]: Unused?

Comment (SimonSchick) [reply package.json]: Good catch, I had to use it for a moment, but then removed it!

**Review by connor4312 — APPROVED** [review]
Approving pending dropping cross-env 😄

**Comment by SimonSchick** [comment]
I never merged this >.>

**Pull request merged**